src/plugins/intel_cpu/src/nodes/conv.cpp

```diff
@@ -1481,9 +1500,21 @@ void Convolution::prepareParams() {
 
     if (execPtr) {
         primArgs[DNNL_ARG_SRC] = srcMemPtr->GetPrimitive();
-        primArgs[DNNL_ARG_WEIGHTS] = wghMemPtr->GetPrimitive();
         primArgs[DNNL_ARG_DST] = dstMemPtr->GetPrimitive();
 
+        if (key.constWeight) {
+            // const weight preparation/reordering needs to be done once at next execution
+            // when the input weight data is guaranteed to be ready (considering possible const-folding
+            // subgraphs inserted between constant weight node and conv)
+            auto it = primArgs.find(DNNL_ARG_WEIGHTS);
+            if (it == primArgs.end() || it->second.get_desc() != execPtr->getWeightDesc()) {
```

Comment:
Unfortunately, `get_desc()` is a costly operation since it does memory descriptor clone under the hood. So to reduce the impact, it does make sense to consider the approach with using `prevExecPtr` to read the memory descriptor.

Reply:
Good idea, will change!